torchrl/envs/transforms/transforms.py

```diff
@@ -2433,3 +2433,51 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}(decay={self.decay:4.4f},"
             f"eps={self.eps:4.4f}, keys={self.in_keys})"
         )
+
+
+class RewardSum(Transform):
+    """Tracks the accumulated reward of each episode.
+
+    This transform requires ´reward´ to be input key. If that is not the case,
+    the transform has no effect.
+    """
+
+    inplace = True
+
+    def __init__(
+        self,
+        in_keys: Optional[Sequence[str]] = None,
+        out_keys: Optional[Sequence[str]] = None,
+    ):
+        in_keys = ["reward"]
+        if out_keys is None:
+            out_keys = ["episode_reward"]
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+
+    def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Resets episode rewards."""
+        if "reset_workers" in tensordict.keys():
+            for out_key in self.out_keys:
+                if out_key in tensordict.keys():
+                    tensordict[out_key][tensordict["reset_workers"]] = 0.0
+
+        return tensordict
+
+    def _call(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Updates the episode rewards with the step rewards."""
+        # Sanity checks
+        self._check_inplace()
+        for in_key in self.in_keys:
+            if in_key not in tensordict.keys():
```

Comment:
what if we have 2 keys (e.g. "reward" and "reward2") and only "reward2" is present?

Reply:
If there can only be one "in_keys" we could check that in the constructor? Then we can just do
```
in_key = self.in_keys[0]
out_key = self.out_keys[0]
```
and avoid relying on for loops in the transform

Reply:
Now it handles more that one input key (e.g. "reward" and "reward2") and creates outputs based on these names ("episode_reward" and "episode_reward2")